packages/manager/modules/sms/src/sms/batches/history/telecom-sms-batches-history.controller.js

```diff
@@ -0,0 +1,62 @@
+import get from 'lodash/get';
+import moment from 'moment';
+
+export default class SmsBatchesHistoryController {
+  /* @ngInject */
+  constructor($http, $q, SmsService) {
+    this.$http = $http;
+    this.$q = $q;
+    this.SmsService = SmsService;
+  }
+
+  cancelBatches(selectedBatches) {
+    this.isCanceling = true;
+
+    return this.$q
+      .all(
+        selectedBatches
+          .filter(({ status }) => this.canBeCancelled(status))
```

Comment:
Could it make sense to prevent selecting batches that can't be cancelled ?

Reply:
Hmm I guess no as the rows are selectable through the datagrid, even the ones which can't be cancelled